python/pybind11/cache/cache_py.h

```diff
@@ -40,7 +40,7 @@ py::dict py_config(ImageCache& cache);
 void py_image_cache_reserve(ImageCache& cache, uint32_t memory_capacity, py::kwargs kwargs);
 
 py::int_ py_preferred_memory_capacity(const py::object& img,
-                                      const std::optional<const std::vector<uint32_t>>& image_size,
+                                      const std::optional<const std::vector<uint64_t>>& image_size,
```

Comment:
using `uint32_t` was just a naive thought that input image size (width or height) wouldn't be higher than 2,000,000,000px but it can cause problems in the distant future so I made sure it uses 64-bit type for image size.